src/component/contents/DraftEditorContents.react.js

```diff
@@ -117,109 +143,144 @@ class DraftEditorContents extends React.Component<Props> {
     const directionMap = nullthrows(editorState.getDirectionMap());
 
     const blocksAsArray = content.getBlocksAsArray();
+    const rootBlock = blocksAsArray[0];
     const processedBlocks = [];
-    let currentDepth = null;
-    let lastWrapperTemplate = null;
-
-    for (let ii = 0; ii < blocksAsArray.length; ii++) {
-      const block = blocksAsArray[ii];
-      const key = block.getKey();
-      const blockType = block.getType();
-
-      const customRenderer = blockRendererFn(block);
-      let CustomComponent, customProps, customEditable;
-      if (customRenderer) {
-        CustomComponent = customRenderer.component;
-        customProps = customRenderer.props;
-        customEditable = customRenderer.editable;
-      }
 
-      const {textDirectionality} = this.props;
-      const direction = textDirectionality
-        ? textDirectionality
-        : directionMap.get(key);
-      const offsetKey = DraftOffsetKey.encode(key, 0, 0);
-      const componentProps = {
-        contentState: content,
-        block,
-        blockProps: customProps,
-        customStyleMap,
-        customStyleFn,
-        decorator,
-        direction,
-        forceSelection,
-        key,
-        offsetKey,
-        selection,
-        tree: editorState.getBlockTree(key),
-      };
-
-      const configForType =
-        blockRenderMap.get(blockType) || blockRenderMap.get('unstyled');
-      const wrapperTemplate = configForType.wrapper;
-
-      const Element =
-        configForType.element || blockRenderMap.get('unstyled').element;
-
-      const depth = block.getDepth();
-      let className = this.props.blockStyleFn(block);
-
-      // List items are special snowflakes, since we handle nesting and
-      // counters manually.
-      if (Element === 'li') {
-        const shouldResetCount =
-          lastWrapperTemplate !== wrapperTemplate ||
-          currentDepth === null ||
-          depth > currentDepth;
-        className = joinClasses(
-          className,
-          getListItemClasses(blockType, depth, shouldResetCount, direction),
-        );
+    if (rootBlock && rootBlock instanceof ContentBlockNode) {
```

Comment:
For now left the rendering logic for tree inside `DraftEditorContents.react.js`, but when we introduce a new feature flag to control this logic we can move this out to a tree aware variant of `DraftEditorContents.react.js` and use the feature flag to chose between both.

Decided to integrate in here for now to make it easier to review and get early end to end progression